### PR TITLE
Improve documentation

### PR DESCRIPTION
**Pull request by adamchainz**
1. Upgrade versions on Read the Docs.
2. Use a consistent tagline everywhere that mentions “HTML sanitization”, so the purpose is clearer without having to read about Ammonia.
3. Give examples of every function.
4. Document `ALLOWED_TAGS` and `ALLOWED_ATTRIBUTES`.
5. Improve decsriptions.
6. Minimize requirements.txt

### CODE REVIEW

**Comment by adamchainz** [comment]
Will fix later.

File: docs/index.rst [hunk 1]
```diff
@@ -8,32 +10,53 @@ Installation
...
+
+API reference
 -------------
 
 .. automodule:: nh3
    :members:
 
-.. toctree::
-   :maxdepth: 2
-   :caption: Contents:
```
Comment (adamchainz) [inline docs/index.rst:30]: This is not necessary for a single page of docs.

File: docs/index.rst [hunk 2]
```diff
@@ -8,32 +10,53 @@ Installation
...
+   The default mapping of tags to allowed attributes for ``clean()``.
+   Useful for customizing the default to add or remove some attributes:
 
-Indices and tables
-==================
+   .. code-block:: pycon
 
-* :ref:`genindex`
-* :ref:`modindex`
-* :ref:`search`
```
Comment (adamchainz) [inline docs/index.rst:39]: Similarly unnecessary.

File: docs/requirements.txt [hunk 3]
```diff
@@ -1,27 +1,2 @@
...
-sphinx-basic-ng==1.0.0b1
-sphinxcontrib-applehelp==1.0.4
-sphinxcontrib-devhelp==1.0.2
-sphinxcontrib-htmlhelp==2.0.0
-sphinxcontrib-jsmath==1.0.1
-sphinxcontrib-qthelp==1.0.3
-sphinxcontrib-serializinghtml==1.1.5
-urllib3==1.26.14
+shibuya
+sphinx
```
Comment (adamchainz) [inline docs/requirements.txt:2]: Using unpinned requirements for docs builds is usually fine - I haven't had any issues with even my larger projects.

File: docs/conf.py [hunk 4]
```diff
@@ -39,8 +39,12 @@
 # List of patterns, relative to source directory, that match files and
 # directories to ignore when looking for source files.
 # This pattern also affects html_static_path and html_extra_path.
-exclude_patterns = ['_build', 'Thumbs.db', '.DS_Store']
-
+exclude_patterns = [
+    '.DS_Store',
+    '.venv',
```
Comment (adamchainz) [inline docs/conf.py:44]: Just adding `.venv` to make it easier for local setup.

**Review by messense — APPROVED** [review]

**Pull request merged**